falcon/http_error.py

```diff
@@ -138,6 +127,8 @@ def __repr__(self):
 
     @property
     def has_representation(self):
+        warn = 'has_representation is deprecated and is currently unused by falcon'
+        warnings.warn(warn, DeprecatedWarning)
```

Comment:
In the default CPython configuration, these are unfortunately oftentimes suppressed.

Since under the hood a property is a function, maybe you could just reuse our [`deprecated`](https://github.com/falconry/falcon/blob/master/falcon/util/misc.py#L123) utility?

Reply:
I use `DeprecatedWarning` of falcon.

I'm not using deprecated since it adds `Call to deprecated function has_representation(...)` before the specified warning, and I thought that it may create confusion